src/main/kotlin/com/openlattice/linking/PostgresLinkingLogService.kt

```diff
@@ -73,11 +71,29 @@ class PostgresLinkingLogService( val hds: HikariDataSource ) : LinkingLogService
 
 private val LOG_COLUMNS = LINKING_LOG.columns.joinToString(",", transform = PostgresColumnDefinition::getName)
 
-private val INSERT_LOG_SQL = "INSERT INTO ${LINKING_LOG.name} ($LOG_COLUMNS) VALUES (?,?,?,?)"
+private val INSERT_LOG_SQL = "INSERT INTO ${LINKING_LOG.name} ($LOG_COLUMNS) VALUES (?,?::jsonb,?)"
+
+// Read latest version of a link
+private val READ_LATEST_LINKED_SQL = "SELECT ${ID_MAP.name} FROM ${LINKING_LOG.name} WHERE ${LINKING_ID.name} = ? ORDER BY ${VERSION.name} LIMIT 1"
+
+// Add a link. Grab previous value from a temp table
+private val ADD_LINK_SQL = "WITH old_json as " +
+        "( SELECT ${ID_MAP.name} as value FROM ${LINKING_LOG.name} WHERE ${LINKING_ID.name} = ? ORDER BY ${VERSION.name} LIMIT 1 )" +
+        "UPDATE ${LINKING_LOG.name} " +
+        "SET ${LINKING_ID.name} = ?" +
+        "${ID_MAP.name} = jsonb_set( old_json.value, '{?}', ${ID_MAP.name}->'?' || '?' )," +
+        "${VERSION.name} = ?" +
```

Comment:
Some spaces are missing from here

Reply:
Good catch! I was just writing some tests and saw that!